omim2obo/parsers/omim_txt_parser.py

```diff
@@ -116,14 +116,107 @@ def convert_txt_to_tsv(file_name: str):
     df.to_csv(mim_file_tsv_path, sep='\t', index=False)
 
 
-def get_mim_file(file_name: str, download=False, return_df=False) -> Union[List[str], pd.DataFrame]:
+# TODO: branch into 2 sep funcs if indeed need to update 2 files
+#  - if only 1 file needed, drop file_name param
+# TODO: do we need to update genemap2? is that even being used?
+#  - if so, need to add here, and in 1 place in get_mim_file()
+def update_mim_file_with_protected(
+    file_name: str, inpath: str, outpath: str, protected_path=DISEASE_GENE_PROTECTED_PATH
+):
+    """Update the files downloaded from OMIM to add information we've set in protected-disease-gene.tsv
+
+    Information about this 'protected' file can be found in the docs (README.md).
+
+    mim2gene.txt: As of 2025/03/05, it turned out there were no missing rows in this file. So this code is here now more
+    for futureproofing purposes.
+    """
+    outpath_with_header = outpath.replace('.tsv', '-with-header.tsv')
+    df = pd.read_csv(inpath, comment='#', sep='\t', dtype=str).fillna('')
+    df['is_added_protection'] = False
+    prot_df = pd.read_csv(protected_path, sep='\t').fillna('')
+
+    # TODO: is this needed by morbidmap? if not, move below
+    hgnc_id_symbols: Dict[str, str] = get_hgnc_id_symbol_map()
+
+    if file_name == 'morbidmap.txt':
+        new_prot_rows = []
```

Comment:
@joeflack4 I pulled the latest change from this morning and then ran `python -m omim2obo` and `sh run.sh make omim.owl` and the entries that should be excluded are still excluded and entries that should be protected are still protected. Are these the only rows changed in the diff after this code change?

Reply:
I [was asking](https://github.com/monarch-initiative/omim/pull/198#discussion_r1984384248) if you could review the diff. I glanced at it, and it seemed like the number of +'s roughly lined up with the number of protections, and the entries I looked at were indeed from the protections file.

If you want to know 100% for sure that there is nothing in the diff at all that is unexpected, that could take a bit of time to asses and I don't know if I have that much time.

Reply:
I reviewed the initial diff, but did not create a new one and was wondering if you could.

Reply:
Oh, I will if you really want me to. Just trying to be economical with time. I have confidence it will be the same, as the only lines that were removed were `prot_df = prot_df[prot_df['type'] == 'digenic']` ([commit](https://github.com/monarch-initiative/omim/pull/198/commits/2887bb9d1b5f4dc53188a736df284934445a03f5)), and all of the entries in the review file have 'type' digenic. So I think we can say with a high degree of confidence it will be OK.

Reply:
I compared the new diff and that also looks good.

Reply:
Thanks!

Reply:
@twhetzel I just merged. This is the only thing different between `develop` and `master`. I'd better run a build on the `omim` branch with this now. Can you do the `mondo-ingest` build when you are ready?

Reply:
Yes, I can do a build on `mondo-ingest`